finalitygadget/finalitygadget.go

```diff
@@ -269,11 +269,22 @@ func (fg *FinalityGadget) QueryBlockRangeBabylonFinalized(
  * returns math.MaxUint64, ErrBtcStakingNotActivated if the BTC staking is not activated
  */
 func (fg *FinalityGadget) QueryBtcStakingActivatedTimestamp() (uint64, error) {
```

Comment:
This function does not look like a query function as it involves DB operations. How about we

- let this function perform a query only
- move the process of saving the activated timestamp somewhere else (e.g., upon initialisation of finality gadget)